app/Services/Training/CheckWaitingListEligibility.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace App\Services\Training;
+
+use App\Models\Mship\Account;
+use App\Models\NetworkData\Atc;
+use App\Models\Training\WaitingList;
+use Carbon\Carbon;
+
+class CheckWaitingListEligibility
+{
+    public function __construct(
+        private Account $account
+    ) {
+    }
+
+    private ?bool $baseControllingHoursCheck = null;
+
+    private ?bool $waitingListFlagsCheck = null;
+
+    public function checkBaseControllingHours(WaitingList $waitingList)
+    {
+        if (! $waitingList->should_check_atc_hours || $waitingList->department == WaitingList::PILOT_DEPARTMENT) {
+            $this->baseControllingHoursCheck = true;
+
+            return true;
+        }
+
+        // avoid extra queries if method called multiple times
+        // for lifecycle of class.
+        if ($this->baseControllingHoursCheck !== null) {
+            return $this->baseControllingHoursCheck;
+        }
+
+        $recentAtcMinutes = Atc::where('account_id', $this->account->id)
+            ->where('disconnected_at', '>=', Carbon::parse('3 months ago'))->isUk()
+            ->sum('minutes_online');
+
+        $this->baseControllingHoursCheck = $recentAtcMinutes >= 720;
+
+        return $this->baseControllingHoursCheck;
+    }
+
+    public function checkWaitingListFlags(WaitingList $waitingList)
+    {
+        $waitingListAccount = $waitingList->accounts()->where('account_id', $this->account->id)->first()->pivot;
```

Comment:
Shouldn't `account_id` be `id` here because it's querying `App\Models\Mship\Account` models not `WaitingListAccount` pivot models? I guess it is using the pivot model attribute in this case, but might be best to stick to `id` for convention?

Reply:
Sadly it's ambiguous so account_id is needed